pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -138,11 +138,31 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 			attrs.linkTypeP2 = EthLinkType
 			needReboot = true
 		}
-		if needReboot {
+		if needReboot || changeWithoutReboot {
 			attributesToChange[ifaceSpec.PciAddress] = *attrs
 		}
 	}
 
+	// Set total VFs to 0 for mellanox interfaces with no spec
+	for pciPrefix := range mellanoxNicsStatus {
+		if _, ok := processedNics[pciPrefix]; ok {
+			continue
+		}
+
+		// Add the nic to processed Nics to not repeat the process for dual nic ports
+		processedNics[pciPrefix] = true
+		pciAddress := pciPrefix + "0"
+
+		_, fwNext, err := getMlnxNicFwData(pciAddress)
+		if err != nil {
+			return false, false, err
+		}
+
+		if fwNext.totalVfs > 0 || fwNext.enableSriov {
+			attributesToChange[pciAddress] = mlnxNic{totalVfs: 0}
```

Comment:
@Mmduh-483 please add a level 2 log here: `glog.V(2).Infof("Changing TotalVfs %d to 0, don't require rebooting", fwNext.totalVfs)`